### ISSUE TITLE
Setup CLI functionality for i18n-check

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/i18n-check#4]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/i18n-check/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](.github/CODE_OF_CONDUCT.md)

### Description

The basic idea of `i18n-check` is that it will function as a CLI. We should leverage `argparse` to bring in arguments for the CLI, but the main usage of the package will not be argument based. Rather we want the user to be able to do the following in the command line:

```bash
i18n-check
```

From here the CLI would check an `i18n_check_config.yaml` file for the arguments needed for the checks, which would then be ran. Arguments for the CLI would be:

- `-h` / `--help` (we'd need to change the first letter of the help message to being capitalized)
- `-v` / `--version` (check the version of the CLI)
- `-u` / `--upgrade` (upgrades to the latest version)
- `-cc` / `--create-config` (an interactive prompt to then create a config file at the end)
    - This would include placeholders to suggest common structures

The `-cc` functionality would also automatically be ran if the user tries to use the check without a config file.

### Contribution

Happy to discuss this and review the work when a PR is opened! 🚀

**Comment by andrewtavis** [comment]
CC @OmarAI2003 👋 Let me know if there are questions on the above! :) Great to have you in the sync! 😊

**Comment by andrewtavis** [comment]
I think that these changes can go in a `scr/i18n-check/cli.py` file, and we can follow the `src/scribe_data/cli/main.py` for this :)

**Comment by OmarAI2003** [comment]
> CC @OmarAI2003 👋 Let me know if there are questions on the above! :) Great to have you in the sync! 😊

Hi @andrewtavis ,of course! I've got a lot of questions, and I'm excited to start on this. However, I have three exams left, so I’ll begin working on this seriously by the 7th of January.

Also, ideally, I should do the test related #2 issue before starting this right?

**Comment by andrewtavis** [comment]
You can work on this or #2 beforehand, @OmarAI2003 :) Whatever works best for you 😊 Let's check in in the new year after your tests, and good luck on them! 😊

**Comment by andrewtavis** [comment]
Quick note here while I'm thinking about it, @OmarAI2003 :) We should try to figure out a way for `directories-to-skip` to allow for some directories to be allowed for some checks and not for others. Maybe some of these should be based on the check level, so the YAML could be something like:

```yaml
src-dir: frontend
i18n-dir: frontend/i18n
i18n-src: frontend/i18n/en.json

checks:
  - all:
    active: true
    skip: [frontend/node_modules]
  - invalid-keys:
    skip: []
  - key-identifiers:
    skip: []
  - unused-keys:
    skip: []
  - non-source-keys:
    skip: []
  - repeat-values:
    skip: []

file-types-to-check: [.ts, .js]
warn-on-nested-i18n-src: true
```

I.e. there would be `all` that would effect all checks for global arguments, and then we could do check level assignments as well? Let me know what you think :) Open to suggestions!

**Comment by andrewtavis** [comment]
Let's get some work done here now, @OmarAI2003, and then I can open another testing issues for the rest of it once we have the context here finalized :) Happy to discuss the work for this issue in more detail, @OmarAI2003! Please let me know if there are any questions 😊

**Comment by andrewtavis** [comment]
Here's [src/scribe_data/cli/main.py](https://github.com/scribe-org/Scribe-Data/blob/main/src/scribe_data/cli/main.py) again for us to reference in creating the functionality here, @OmarAI2003 :)

Would be great if we could start with a PR for the following args:

- `-h` / `--help` (we'd need to change the first letter of the help message to being capitalized)
- `-v` / `--version` (check the version of the CLI)
- `-u` / `--upgrade` (upgrades to the latest version)

From there we can do individual arguments for each check, and then another PR for the `-cc` / `--create-config` arg and running the whole CLI with:

```
i18n-check
```

I'm going to simplify the check directory file names really quick right now :)

**Comment by andrewtavis** [comment]
So once the above first three arguments are done we can do another PR that adds arguments for each file in [src/i18n_check/check](https://github.com/activist-org/i18n-check/tree/main/src/i18n_check/check):

- `all`
- `invalid-keys`
- `key-identifiers`
- `nested-i18n-src`
- `non-source-keys`
- `repeat-values`
- `unused-keys`

**Comment by OmarAI2003** [comment]
> Quick note here while I'm thinking about it, [@OmarAI2003](https://github.com/OmarAI2003) :) We should try to figure out a way for `directories-to-skip` to allow for some directories to be allowed for some checks and not for others. Maybe some of these should be based on the check level, so the YAML could be something like:

This approach looks perfect to me. We also may want to consider adding 
```yaml 
 directories-to-skip-global: [frontend/node_modules]
``` 
if some directories need to be skipped across all checks.

**Comment by andrewtavis** [comment]
Nice ok :) Let's start on the first parts of the functionality and first arguments then 😊

**Comment by andrewtavis** [comment]
First processes are up and running, @OmarAI2003 😊 Do we want to start on the commands mentioned [here](https://github.com/activist-org/i18n-check/issues/4#issuecomment-2689352437) now? Let me know if you have any questions! :)

**Comment by andrewtavis** [comment]
A final thing that's needed to be done here, @OmarAI2003, is that we want to add tests of the actual checks such that we'll go through and assert that each check fails for the conditions we've set within the test files. Let me know if you have interest in looking into this! Maybe we can make a new issue for this?

**Comment by andrewtavis** [comment]
I'd say nothing else, @OmarAI2003 :) Please make the other issue 😊

**Comment by andrewtavis** [comment]
And thanks for the great work here! 🚀
